### PR TITLE
scsi_id: use read_line instead of fgets

### PR DESCRIPTION
**Pull request by bluca**
LGTM warns about it:

"Call to fgets() is potentially dangerous. Use read_line() instead."

Missed this in the previous series. This is the last error on LGTM.com, the rest are all notices about `// FIXME`

### CODE REVIEW

File: src/udev/scsi_id/scsi_id.c [hunk 1]
```diff
@@ -136,12 +137,13 @@ static int get_file_options(const char *vendor, const char *model,
 
                 vendor_in = model_in = options_in = NULL;
 
-                buf = fgets(buffer, MAX_BUFFER_LEN, f);
-                if (!buf)
+                r = read_line(f, MAX_BUFFER_LEN, &buffer);
```
Comment (yuwata) [inline src/udev/scsi_id/scsi_id.c:130]: Hmm? `read_line()` allocates the memory, so the buffer allocated in the above is leaked...

**Review by yuwata — CHANGES_REQUESTED** [review]

File: src/udev/scsi_id/scsi_id.c (hunk 1 above)
Comment (bluca) [reply src/udev/scsi_id/scsi_id.c:130]: agh of course, I don't know what I was thinking - fixed

**Comment by yuwata** [comment]
LGTM.

**Comment by bluca** [comment]
> TEST-17-UDEV:                       FAIL     (200 s)

**Pull request merged**